frontend/packages/console-dynamic-plugin-sdk/src/runtime/plugin-api.ts

```diff
@@ -0,0 +1,13 @@
+import * as _ from 'lodash';
+
+import {
+  useK8sWatchResource,
+  useK8sWatchResources,
+} from '@console/internal/components/utils/k8s-watch-hook';
+
+export const exposePluginAPI = _.once(() => {
+  window.api = {
+    useK8sWatchResource,
+    useK8sWatchResources,
+  };
+});
```

Comment:
IMO we should be using webpack module federation for exposing API instead of externals and having to create a global API object.

Plugins who want to define their own API for other plugins will need a way to expose their API and that should be consistent with the console core.